Role: Coder-R226

Updated this branch with commit cdc8e4096.

Change scope:
- in gtest_tiforth_execution_host_v2_inner_hash_join.cpp, the two high-partition outer-join "legacy end" slices now truly exercise legacy end-of-input by setting build_end_with_output=false and probe_end_with_output=false
- parity log marker updated from end_with_output=1 to end_with_output=0 in those slices

Validation in this round:
- cargo build -p tiforth-ffi-c
- linked donor configure in /tmp/tiflash-r226-issue252-linked-host-v2 with USE_INTERNAL_LZ4_LIBRARY=OFF
- full build attempt for gtests_tiforth_execution_host_v2 progressed through compile of the changed proving TU and stopped at final link with existing environment mismatch: Undefined symbols for architecture arm64: _EVP_sm4_ctr
- no new strict runtime-pass claim in this round

--- a/dbms/src/Functions/tests/gtest_tiforth_execution_host_v2_inner_hash_join.cpp
+++ b/dbms/src/Functions/tests/gtest_tiforth_execution_host_v2_inner_hash_join.cpp
@@ -1464,6 +1464,50 @@ TEST_F(TestTiforthExecutionHostV2InnerHashJoin, InnerHashJoinPayloadParityHighPa
               << " donor_rows=" << donor_serial.rows.size() << " max_block_size=1 parity=ok" << std::endl;
 }
 
+TEST_F(
+    TestTiforthExecutionHostV2InnerHashJoin,
+    InnerHashJoinPayloadParityHighPartitionMaxBlockLegacyEndSerialAndParallel)
+{
+    auto donor_serial = runDonorNativeInnerJoin(1);
+    auto donor_parallel = runDonorNativeInnerJoin(4);
+
+    ASSERT_EQ(donor_serial.warning_count, donor_parallel.warning_count);
+    ASSERT_EQ(donor_serial.rows, donor_parallel.rows);
+
+    AdapterRunResult adapter_serial;
+    runAdapterInnerJoin(
+        8,
+        BATCH_OWNERSHIP_BORROW_WITHIN_CALL,
+        defaultInnerJoinBuildRows(),
+        defaultInnerJoinProbeRows(),
+        1,
+        false,
+        false,
+        adapter_serial);
+    AdapterRunResult adapter_parallel;
+    runAdapterInnerJoin(
+        8,
+        BATCH_OWNERSHIP_FOREIGN_RETAINABLE,
+        defaultInnerJoinBuildRows(),
+        defaultInnerJoinProbeRows(),
+        1,
+        false,
+        false,
+        adapter_parallel);
+
+    ASSERT_EQ(adapter_serial.warning_count, donor_serial.warning_count);
+    ASSERT_EQ(adapter_parallel.warning_count, donor_serial.warning_count);
+    ASSERT_EQ(adapter_serial.rows, donor_serial.rows);
+    ASSERT_EQ(adapter_parallel.rows, donor_serial.rows);
+
+    std::cout << "[tiforth-host-v2-inner-join-high-partition-legacy-end] serial=8 warnings="
+              << adapter_serial.warning_count << " rows=" << adapter_serial.rows.size()
+              << " parallel=8 warnings=" << adapter_parallel.warning_count
+              << " rows=" << adapter_parallel.rows.size() << " donor_warnings=" << donor_serial.warning_count
+              << " donor_rows=" << donor_serial.rows.size() << " max_block_size=1 end_with_output=0 parity=ok"
+              << std::endl;
+}
+
 TEST_F(TestTiforthExecutionHostV2InnerHashJoin, InnerHashJoinPayloadFanoutParityHighPartitionMaxBlockSerialAndParallel)
 {
     auto donor_serial = runDonorNativeInnerJoinFanout(1);

